### PR TITLE
Chore/extract api package

### PR DESCRIPTION
**Pull request by en0ma**
This PR moves all API logic into an `apiV1` package and sets the ground for API v2 work which is `apiv2` both in the `api` directory. It is also a step toward improving the code for readability, testing, and identifying service boundaries (which will help us build a better deterministic platform).

**verification**
- I have tested `content/add`, `content/add-car`, `pinnging/pin`, `staging-zones`, and `deal-making`, `shuttles` - all works

Please have a look and provide feedback

### CODE REVIEW

**Review by gmelodie — COMMENTED** [review]
It makes very little sense to me to add the API changes alongside other refactors on the same PR. Can you decouple these into two or more PRs? If so I'll approve the API changes

**Comment by en0ma** [comment]
> It makes very little sense to me to add the API changes alongside other refactors on the same PR. Can you decouple these into two or more PRs? If so I'll approve the API changes

I totally understand @gmelodie, But some of those changes are required to bring the API abstraction to life. I know there are some changes that could have been avoided mostly around the content manager (moving files around - I needed to do that as I tried to solve the pinner cyclic dependence error)

**Review by gmelodie — APPROVED** [review]

**Comment by alvin-reyes** [comment]
Let’s not merge this until we get v0.2.1 out.

**Comment by gmelodie** [comment]
@en0ma since this is blocking APIv2 development, let's merge it to `api-v2`

File: contentmgr/peice_commp.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
+package contentmgr
```
Comment (neelvirdy) [inline contentmgr/peice_commp.go:1]: nit: filename spelling > `piece_commp.go`

File: api/v1/handlers.go [hunk 2]
```diff
@@ -3902,7 +3654,7 @@ func (s *Server) handleDeleteContentFromCollection(c echo.Context, u *util.User)
 	return c.NoContent(http.StatusOK)
 }
 
-func (s *Server) tracingMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+func (s *apiV1) tracingMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
```
Comment (neelvirdy) [inline api/v1/handlers.go:3678]: you can remove this since you have it in util/tracing_middleware.go and configure it in api_engine already

**Comment by elijaharita** [comment]
microscopic nit: could we bring the package names more inline with the go style guide? https://go.dev/blog/package-names

my recommendations: 
`api_engine` -> `apiengine` or `apieng`
`apiV1` -> `apiv1` or just `api` (if  two versions are imported, they can be renamed locally at import to `apiv1`, `apiv2`, etc.)

File: pinner/progress/progress.go [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package pinning_progress
```
Comment (neelvirdy) [inline pinner/progress/progress.go:1]: is this worth having as its own package? or own file? seems like declaring a type in pinmgr.go would be ok, or we could have a types.go file in the pinner package to declare this

File: pinner/pinning_op/pinning_op.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package pinning_op
```
Comment (neelvirdy) [inline pinner/pinning_op/pinning_op.go:1]: also wondering why this has to be its own package, seems more convenient to have it be part of the pinner package

File: pinner/pinning_op/pinning_op.go [hunk 5]
```diff
@@ -36,24 +36,42 @@ type PinningOperation struct {
...
 	po.LastUpdate = time.Now()
 }
+
+func (po *PinningOperation) Complete() {
+	po.lk.Lock()
+	defer po.lk.Unlock()
+
+	po.EndTime = time.Now()
+	po.LastUpdate = time.Now()
+	po.Status = types.PinningStatusPinned
```
Comment (neelvirdy) [inline pinner/pinning_op/pinning_op.go:68]: nit: can we update the status before these timestamps and use the same timestamp for EndTime and LastUpdate?

File: pinner/pinning_op/pinning_op.go [hunk 6]
```diff
@@ -36,24 +36,42 @@ type PinningOperation struct {
...
 
 func (po *PinningOperation) Fail(err error) {
-	po.Lk.Lock()
+	po.lk.Lock()
+	defer po.lk.Unlock()
+
 	po.FetchErr = err
 	po.EndTime = time.Now()
 	po.Status = types.PinningStatusFailed
 	po.LastUpdate = time.Now()
```
Comment (neelvirdy) [inline pinner/pinning_op/pinning_op.go:51]: nit: can we update the status before these timestamps and use the same timestamp for EndTime and LastUpdate?

**Comment by neelvirdy** [comment]
another nit: it would be nice if we could have subfolders, packages, and API groups match when feasible.

example: `pinner/` -> `pinning/` for folder name, contains a `pinning` package, implements the `/pinning` API group.

miners may be another feasible case, but totally fair to include that with the storage providers switch later in API v2

File: pinner/progress/progress.go (hunk 3 above)
Comment (neelvirdy) [reply pinner/progress/progress.go:1]: nvm - i see the cycle with cm now

**Comment by en0ma** [comment]
> another nit: it would be nice if we could have subfolders, packages, and API groups match when feasible.
> 
> example: `pinner/` -> `pinning/` for folder name, contains a `pinning` package, implements the `/pinning` API group.
> 
> miners may be another feasible case, but totally fair to include that with the storage providers switch later in API v2

Yeah, this will come in a different PR, I did not want to make too much unrelated changes. Ideally, the pinner package should hold all pinning logic. This PR is a first step to better abstractions.

**Comment by en0ma** [comment]
> microscopic nit: could we bring the package names more inline with the go style guide? https://go.dev/blog/package-names
> 
> namely the no underscores rule and no capitals rule
> 
> my recommendations: `api_engine` -> `apiengine` or `apieng` `apiV1` -> `apiv1` or just `api` (if two versions are imported, they can be renamed locally at import to `apiv1`, `apiv2`, etc.)

Nice, done.

File: pinner/pinning_op/pinning_op.go (hunk 4 above)
Comment (neelvirdy) [inline pinner/pinning_op/pinning_op.go:1]: nit: `pinning_op` -> `operation` for idiomatic package name and so it can be imported as `/pinner/operation` and renamed locally as oneof: `op` or `po` or `pinop`

also should update folder and filename to operation

File: pinner/progress/progress.go (hunk 3 above)
Comment (neelvirdy) [inline pinner/progress/progress.go:1]: nit: `pinning_progress` -> `progress` for idiomatic package name and so it can be imported as `/pinner/progress`

**Comment by neelvirdy** [comment]
LGTM. also manually verified i'm still able to upload, consolidate, and aggregate a deal with an API node + one shuttle setup

**Review by neelvirdy — APPROVED** [review]
